src/main/java/org/opensearch/flowframework/workflow/ProcessNode.java

```diff
@@ -139,51 +141,53 @@ public TimeValue nodeTimeout() {
      * @return this node's future.
      * This is returned immediately, while process execution continues asynchronously.
      */
-    public CompletableFuture<WorkflowData> execute() {
+    public PlainActionFuture<WorkflowData> execute() {
         if (this.future.isDone()) {
             throw new IllegalStateException("Process Node [" + this.id + "] already executed.");
         }
 
         CompletableFuture.runAsync(() -> {
```

Comment:
In fact I don't have preference between `CompletableFuture` and `PlainActionFuture`, the bottleneck I meant is that the runAsync or other run methods in CompletableFuture uses ForkjoinPool which could be a small number, we should provide a customer thread pool for CompletableFuture to run tasks, e.g. `CompletableFuture.runAsync(() -> {yourTask}, Executors.myExcutor())`

Reply:
We do use our own thread pool executor for the [two times we use `runAsync()`](https://github.com/search?q=repo%3Aopensearch-project%2Fflow-framework%20runasync&type=code)
https://github.com/opensearch-project/flow-framework/blob/bf2cad068996c4aee9f38352ef3dab4a04c1823b/src/main/java/org/opensearch/flowframework/workflow/ProcessNode.java#L188

and 

https://github.com/opensearch-project/flow-framework/blob/bf2cad068996c4aee9f38352ef3dab4a04c1823b/src/main/java/org/opensearch/flowframework/workflow/AbstractRetryableWorkflowStep.java#L131

Are there any other bottlenecks you see?

The main benefit of this replacement, as I understood it, is removing (blocking) `.join()` or `.get()` calls from `CompletableFuture` with their equivalent methods in `PlainActionFuture` which extends `ActionListener` and thus uses the `onResponse` and `onFailure` callbacks which I presume work much better in a multithreaded/multichannel environment.

Reply:
@zane-neo any other suggestion?